src/sync/app.hpp

```diff
@@ -269,6 +269,23 @@ class App {
      * Log out the given user if they are not already logged out.
      */
     void log_out(std::shared_ptr<SyncUser> user, std::function<void(Optional<AppError>)> completion_block) const;
+    
+    /// Switches the active user to the user with the specified id. The user must
+    /// exist in the list of all users who have logged into this application, and
+    /// the user must be currently logged in, otherwise this will return an
+    /// AppError.
+    ///
+    /// @param user_id A String specifying the desired `user_id`
+    /// @param completion_block will return a SyncUser on success, an AppError on failure
+    void switch_user(const std::string& user_id,
```

Comment:
Why use `user_id` instead of the accepting the `shared_ptr<SyncUser>`? If an SDK wants to Switch a user, the developer would already have a handle to that user which is in the form of a shared_ptr. This kind of API just seems to result in us needing to look things up more times?

Reply:
`shared_ptr<SyncUser>` is now the default parameter